src/test/java/com/google/cloud/tools/appengine/cloudsdk/serialization/CloudSdkVersionTest.java

```diff
@@ -56,6 +57,18 @@ public void testToString() {
   public void testEquals_differentSizes() {
     assertTrue(new CloudSdkVersion("0.1.0").equals(new CloudSdkVersion("0.1")));
     assertTrue(new CloudSdkVersion("0").equals(new CloudSdkVersion("0.0.0.0")));
+    assertFalse(new CloudSdkVersion("1.1").equals(new CloudSdkVersion("1.10")));
+  }
+
+  @Test
+  public void testEquals_preRelease() {
```

Comment:
If we do want to blow up on some kinds of input we should have tests to verify that.